### PR TITLE
Improve documentation of Hypervisor selection algorithm methods

### PR DESCRIPTION
**Pull request by kofrezo**
Try to make the calculation of the performance value used by IGVM to
find a (better) hypervisor more clear by using different wording and add
detailed documentation.

### CODE REVIEW

File: igvm/hypervisor.py [hunk 1]
```diff
@@ -829,92 +829,104 @@ def device_to_netcat(self, device, size, listener):
...
-            hv_cpu_util_dest,
-            hv_cpu_util_log,
+        vm_cpu_usage = self.estimate_vm_cpu_usage(vm)
+        hv_cpu_usage = self.dataset_obj['cpu_util_pct']
+
+        # Take into account cpu_util_pct is outdated
+        #
+        # Take into account recent migrations from and to the Hypervisor to
+        # avoid moving too many VMs to the same Hypervisor or discarding the
+        # Hypervisor as candidate because tge cpu_util_pct is not up-to-date
```
Comment (lacunoc) [inline igvm/hypervisor.py:888]: typo

File: igvm/vm.py [hunk 2]
```diff
@@ -857,28 +857,40 @@ def aws_sync(self) -> dict:
...
         """
 
-        # We can't save floats in graphite cache of serveradmin and therefore
-        # divide it by the value we multiplied it before.
-        vm_load_99 = self.dataset_obj['load_99'] / 1000
-        hv_perffactor_src = self.hypervisor.dataset_obj[
-                                'cpu_perffactor'] / 1000
+        # Serveradmin can not handle floats right now so we safe them as
+        # multiple ones of thousand and just divide them here again.
+
```
Comment (lacunoc) [inline igvm/vm.py:880]: Maybe remove the empty l880

**Review by lacunoc — APPROVED** [review]
I had 2 minor comments (typo, empty line), but these aren't mandatory changes to approve this. Good job, improves the understanding a lot.

**Comment by kofrezo** [comment]
test this please

**Comment by kofrezo** [comment]
test this please

**Pull request merged**